### DISCUSSION TITLE
New units using existing units and c

### DISCUSSION THREAD

**Discussion by CodeDrome** [josdejong/mathjs#2280] [Q&A]
Is it possible to create a new unit for a new dimension but using existing units and constants?

I would like to create a unit for wavelengths of light convertible from and to the existing Hertz unit and using the speedOfLight constant to enable conversions such as:

`const redFrequency = math.unit(385, 'terahz');`
`const redWavelength = redFrequency.to('nanometres'); // expected result approx 780`

and vice versa. Basically we need to divide speedOfLight by hertz, both of which already exist in math.js but don't seem usable within a new unit definition.

This is just an example and I am sure there are many more potential uses.

I tried a few things which I won't bother showing here because they didn't work, but it seems to be only possible to create new units convertible to/from units measuring the same quantity.

Thank you, Chris.

**Comment by josdejong** [comment]
@ericman314 do you have any thoughts on this?

**Reply by ericman314** [reply]
Hmm, I don't think there's any way around it, you'll have to either 1) explicitly divide or multiply by the speed of light to make that conversion, 2) override the existing Hz unit and redefine it to have units of length, or 3) redefine nanometers to have units of time^-1. 

I think options 2 and 3 could land you in trouble if your application has to support anything other than this specific conversion, so my recommendation is option 1, to keep everything perfectly clear what's going on. 

If you need a single function that can perform this conversion, perhaps you could extend the `.to` method (@josdejong, you would know best how to do this) so that it checks if one unit is length and the other is frequency, and if so it divides by the speed of light and then calls the original `to` method.

**Reply by josdejong** [reply]
Thanks for your inputs Eric. You could of course also create a simple conversion function instead of extending the `.to` method, if that makes life easier :)

**Reply by CodeDrome** [reply]
Thanks everyone for your input. It's easy enough to carry out the calculations but I thought it would be worth trying to do it within Math.js.
